src/varname.jl

```diff
@@ -98,6 +106,21 @@ function Base.show(io::IO, vn::VarName)
     end
 end
 
+function Base.show(io::IO, vn::VarName{<:Any, <:Lens})
```

Comment:
There's something off with `DynamicIndexLens`:

```julia
julia> @varname x[1][end]
x[1]Setfield.DynamicIndexLens{var"#3#4"}(var"#3#4"())(_)
```

Reply:
That's just my weird show impl below since Setfield.jl doesn't have a `print_application` (which is what we want here) for `DynamicIndexLens`. I didn't give it too much thought since we're not going to be using `DynamicLens` in DPPL anyways, i.e. we will always concretize.